nuxeo-java-client/src/main/java/org/nuxeo/client/api/objects/Document.java

```diff
@@ -300,11 +307,21 @@ public void setLastModified(String lastModified) {
     }
 
     public void setProperties(Map<String, Object> properties) {
-        this.properties = properties;
+        if (properties == null) {
+            this.properties = new HashMap<>();
```

Comment:
I don't understand why setting the properties to `null` actually create a `new HashMap<>()`?